python/pyiceberg/catalog/hive.py

```diff
@@ -197,17 +234,41 @@ def __init__(self, name: str, properties: Properties, uri: str):
         super().__init__(name, properties)
         self._client = _HiveClient(uri)
 
-    def _convert_hive_into_iceberg(self, table: HiveTable) -> Table:
-        # Requires reading the manifest, will implement this in another PR
-        # Check the table type
-        return Table(identifier=(table.dbName, table.tableName))
+    def _convert_hive_into_iceberg(self, table: HiveTable, metadata_location: Optional[str] = None) -> Table:
+        properties: Dict[str, str] = table.parameters
+        if TABLE_TYPE not in properties or properties[TABLE_TYPE] != ICEBERG:
+            raise NoSuchTableError(f"Table is not an Iceberg table: {table.dbName}.{table.tableName}")
+
+        if not metadata_location:
+            if prop_metadata_location := properties.get(METADATA_LOCATION):
+                metadata_location = prop_metadata_location
+            else:
+                raise NoSuchTableError("Metadata location not found")
+
+        file = PyArrowFile(metadata_location)
+        metadata = FromInputFile.table_metadata(file)
+        return Table(identifier=(table.dbName, table.tableName), metadata=metadata, metadata_location=metadata_location)
+
+    def _write_metadata(self, metadata: TableMetadataV2, metadata_path: str):
+        output_file = PyArrowFile(metadata_path)
+        # 😢 https://github.com/apache/iceberg/pull/5439
+        os.makedirs(f"{metadata.location}metadata".lstrip("file:"), exist_ok=True)
+        ToOutputFile.table_metadata(metadata, output_file)
+
+    def _resolve_table_location(self, location: Optional[str], database_name: str, table_name: str):
+        if not location:
+            database_properties = self.load_namespace_properties(database_name)
+            if database_location := database_properties.get(LOCATION):
+                database_location = database_location.rstrip("/")
+                return f"{database_location}/{table_name}/"
+        raise ValueError("Cannot determine location from warehouse, please provide an explicit location")
```

Comment:
This should respect the `warehouse` property used to construct the catalog rather than failing with `ValueError`

Reply:
Nice, added it, including a test